src/components/TaskModal/TaskModal.tsx

```diff
@@ -0,0 +1,220 @@
+                        <TextField
+                            fullWidth
+                            select
+                            name="project"
+                            label="Project"
+                            onChange={handleChange}
+                            value={project}
+                        >
+                            <MenuItem key={'Project#1'} value={'Project#1'}>
+                                {'Project#1'}
+                            </MenuItem>
+                            <MenuItem key={'Project#2'} value={'Project#2'}>
+                                {'Project#2'}
+                            </MenuItem>
+                        </TextField>
+                    </Grid>
+                    <Grid item xs={6}>
+                        <TextField
+                            fullWidth
+                            InputLabelProps={{ shrink: true }}
+                            name="deadline"
+                            label="Deadline"
+                            type="date"
+                            onChange={handleChange}
+                            value={deadline}
+                        />
+                    </Grid>
+                    <Grid item xs={6}>
+                        <TextField
+                            fullWidth
+                            select
+                            name="assignee"
+                            label="Assignee"
+                            onChange={handleChange}
+                            value={assignee}
+                        >
+                            <MenuItem key={TEST_USER} value={TEST_USER}>
+                                {TEST_USER}
+                            </MenuItem>
+                            <MenuItem key={'Random'} value={'Random'}>
+                                {'Random'}
+                            </MenuItem>
+                        </TextField>
+                    </Grid>
+                    <Grid item xs={6}>
+                        <TextField
+                            fullWidth
+                            required
+                            select
+                            name="priority"
+                            label="Priority"
+                            onChange={handleChange}
+                            value={priority}
+                        >
+                            <MenuItem key={'Low'} value={'Low'}>
+                                {'Low'}
+                            </MenuItem>
+                            <MenuItem key={'Medium'} value={'Medium'}>
+                                {'Medium'}
+                            </MenuItem>
+                            <MenuItem key={'High'} value={'High'}>
+                                {'High'}
+                            </MenuItem>
+                        </TextField>
+                    </Grid>
+                    <Grid item xs={6}>
+                        <Button variant="outlined" onClick={closeAndResetForm}>
```

Comment:
If you conditionally render the modal you don't need to manually reset the form as the internal state is cleared when the component is unmounted